(Draft Ballot): Cleanups and Clarifications (SC30 and SC31 version) (#207)
cabforum/documents: docs/BR.md

**Base Domain Name**: The portion of an applied-for FQDN that is the first domain name node left of a registry-controlled or public suffix plus the registry-controlled or public suffix (e.g. "example.co.uk" or "example.com"). For FQDNs where the right-most domain name node is a gTLD having ICANN Specification 13 in its registry agreement, the gTLD itself may be used as the Base Domain Name.

> **CAA**: From RFC 8659 (<http://tools.ietf.org/html/rfc8659>): "The Certification Authority Authorization (CAA) DNS Resource Record allows a DNS domain name holder to specify the Certification Authorities (CAs) authorized to issue certificates for that domain. Publication of CAA Resource Records allows a public Certification Authority to implement additional controls to reduce the risk of unintended certificate mis-issue."

Reviewer: ```suggestion
**CAA**: From RFC 8659 (<http://tools.ietf.org/html/rfc8659>): "The Certification Authority Authorization (CAA) DNS Resource Record allows a DNS domain name holder to specify the Certification Authorities (CAs) authorized to issue certificates for that domain name. Publication of CAA Resource Records allows a public CA to implement additional controls to reduce the risk of unintended certificate mis-issue."
```

Author: ```suggestion
**CAA**: From RFC 8659 (<http://tools.ietf.org/html/rfc8659>): "The Certification Authority Authorization (CAA) DNS Resource Record allows a DNS domain name holder to specify one or more Certification Authorities (CAs) authorized to issue certificates for that domain name.  CAA Resource Records allow a public CA to implement additional controls to reduce the risk of unintended certificate mis-issue."
```

Thanks for pointing out I'd missed the copy/paste update. Above is the corrected text :)

Reviewer: The original text for 6844 came from section 1, not the abstract. Likewise, my suggested text is from section 1 of 8659.

That being said, the abstract and the intro (section 1) say pretty much the same thing so I don't think it matters much which one is used.

#### 3.2.2.8. CAA Records

As part of the issuance process, the CA MUST check for CAA records and follow the processing instructions found, for each dNSName in the subjectAltName extension of the certificate to be issued, as specififed in RFC 8659. If the CA issues, they MUST do so within the TTL of the CAA record, or 8 hours, whichever is greater.

This stipulation does not prevent the CA from checking CAA records at any other time.

When processing CAA records, CAs MUST process the issue, issuewild, and iodef property tags as specified in RFC 8659, although they are not required to act on the contents of the iodef property tag. Additional property tags MAY be supported, but MUST NOT conflict with or supersede the mandatory property tags set out in this document. CAs MUST respect the critical flag and not issue a certificate if they encounter an unrecognized property tag with this flag set.

> RFC 8659 requires that CAs "MUST NOT issue a certificate unless either (1) the certificate request is consistent with the applicable CAA Resource Record set or (2) an exception specified in the relevant Certificate Policy or Certification Practices Statement applies." For issuances conforming to these Baseline Requirements, CAs MUST NOT rely on any exceptions specified in their CP or CPS unless they are one of the following:

Reviewer: ```suggestion
RFC 8659 requires that CAs "MUST NOT issue a certificate unless the CA determines that either (1) the certificate request is consistent with the applicable CAA RRset or (2) an exception specified in the relevant CP or CPS applies." For issuances conforming to these Baseline Requirements, CAs MUST NOT rely on any exceptions specified in their CP or CPS unless they are one of the following:
```

In all cases, the CA SHALL:

1. generate the CA Key Pair in a physically secured environment as described in the CA's Certificate Policy and/or Certification Practice Statement;
2. generate the CA Key Pair using personnel in Trusted Roles under the principles of multiple person control and split knowledge;
3. generate the CA Key Pair within cryptographic modules meeting the applicable technical and business requirements as disclosed in the CA's Certificate Policy and/or Certification Practice Statement;
4. log its CA Key Pair generation activities; and
5. maintain effective controls to provide reasonable assurance that the Private Key was generated and protected in conformance with the procedures described in its Certificate Policy and/or Certification Practice Statement and (if applicable) its Key Generation Script.

#### 6.1.1.2 RA Key Pair Generation

#### 6.1.1.3 Subscriber Key Pair Generation
The CA SHALL reject a certificate request if one or more of the following conditions are met:

1. The Key Pair does not meet the requirements set forth in Section 6.1.5 and/or Section 6.1.6;
2. There is clear evidence that the specific method used to generate the Private Key was flawed;

> 3. The CA is aware of a demonstrated or proven method that exposes the Subscriber's Private Key to compromise;

Reviewer: The cert hasn't issued yet, so probably we want to change "Subscriber" to "Applicant" in this section.

Author: Good point!

```suggestion
3. The CA is aware of a demonstrated or proven method that exposes the Applicant's Private Key to compromise;
```

> 4. The CA has previously been made aware that the Subscriber's Private Key has suffered a Key Compromise, such as through the provisions of Section 4.9.1.1;

Reviewer: ```suggestion
4. The CA has previously been made aware that the Applicant's Private Key has suffered a Key Compromise, such as through the provisions of Section 4.9.1.1;
```

> 5. The CA is aware of a demonstrated or proven method to easily compute the Subscriber's Private Key based on the Public Key (such as a Debian weak key, see https://wiki.debian.org/SSLkeys).

Reviewer: ```suggestion
5. The CA is aware of a demonstrated or proven method to easily compute the Applicant's Private Key based on the Public Key (such as a Debian weak key, see https://wiki.debian.org/SSLkeys).
```

a. For each dNSName in permittedSubtrees, the CA MUST confirm that the Applicant has registered the dNSName or has been authorized by the domain registrant to act on the registrant's behalf in line with the verification practices of section 3.2.2.4.
b. For each iPAddress range in permittedSubtrees, the CA MUST confirm that the Applicant has been assigned the iPAddress range or has been authorized by the assigner to act on the assignee's behalf.

> c. For each DirectoryName in permittedSubtrees the CA MUST confirm the Applicants and/or Subsidiary's Organizational name and location such that end entity certificates issued from the subordinate CA Certificate will be in compliance with section 7.1.2.4 and 7.1.2.5.

Reviewer: ```suggestion
c. For each DirectoryName in permittedSubtrees, the CA MUST confirm the Applicant's and/or Subsidiary's Organizational name and location such that end entity certificates issued from the subordinate CA Certificate will be in compliance with section 7.1.2.4 and 7.1.2.5.
```

Reviewer: I think the intent here is to specify Affiliate, not necessarily "Subsidiary" specifically.

Author: I'm not sure? We should probably discuss more on list, but we don't really use Affiliate outside the context of either Affiliated with the CA or the case of Enterprise RAs and the O values. I can see why it might make sense, in the context of nameConstraints, to allow enumerating the Affiliates as an alternative to the Enterprise RA path, but have we seen that use case come up yet? I'd be inclined towards tackling it when we've got something concrete.

### 7.1.6 Certificate policy object identifier

This section describes the content requirements for the Root CA, Subordinate CA, and Subscriber Certificates, as they relate to the identification of Certificate Policy.

#### 7.1.6.1. Reserved Certificate Policy Identifiers

> The following Certificate Policy identifiers are reserved for use by CAs as an optional means of asserting that a Certificate complies with these Requirements.

Reviewer 2: Should we clarify that these policy identifiers are meant for end-entity Certificates?

"...of asserting that an end-entity Certificate complies with..."

Author: No, because as discussed on the validation list and GitHub, and this text itself, they are not just that. They also assert the CA complies with these, by definition, since they can be used to issue such certificates.

Reviewer 2: I didn't follow these discussions on GitHub nor the validation SC list. The fact that a CA Certificate asserts one or more of the OIDs that are specifically designed for end-entity certificates (for DV, OV, EV) means that these oids are likely to appear in end-entity certificates, and then, by definition, the CA must comply with the  corresponding Issuer requirements. However, the fact that a CA Certificate contains subject information with and organizationName field and asserts the "DV" policy OID, doesn't mean that it violates the BRs.

Is that clear enough in this ballot?

I'm sorry if this has been answered already.

Prior to including a Reserved Certificate Policy Identifier, the CA MUST ensure the following requirements are met:

* __Certificate Policy Identifier:__ `2.23.140.1.2.1`  
  If the Certificate complies with these requirements and lacks Subject identity information that has been verified in accordance with Section 3.2.2.1 or Section 3.2.3.  
  Such Certificates MUST NOT include `organizationName`, `givenName`, `surname`, `streetAddress`, `localityName`, `stateOrProvinceName`, or `postalCode` in the Subject field.  
* __Certificate Policy Identifier:__ `2.23.140.1.2.2`  
  If the Certificate complies with these Requirements and includes Subject Identity Information that is verified in accordance with Section 3.2.2.1.  
  Such Certificates MUST also include `organizationName`, `localityName` (to the extent such field is required under Section 7.1.4.2.2), `stateOrProvinceName` (to the extent such field is required under Section 7.1.4.2.2), and `countryName` in the Subject field.  
* __Certificate Policy Identifier:__ `2.23.140.1.2.3`  
  If the Certificate complies with these Requirements and includes Subject Identity Information that is verified in accordance with Section 3.2.3.  

>   Such Certificates MUST also include either `organizationName` or both `givenName` and `surname`, `localityName` (to the extent such field is required under Section 7.1.4.2.2), `stateOrProvinceName` (to the extent required under Section 7.1.4.2.2), and `countryName` in the Subject field.  

Reviewer: This reads that O and SN + GN are mutually exclusive for IV certs. My understanding was that all 3 RDNs could be included in the certificate (provided that the O conveys the natural person's name), as not all software supports SN + GN.

Author: Do you have a suggested wording?

The existing wording was:
> If the Certificate asserts the policy identifier of 2.23.140.1.2.3, then it MUST also include (i) either `organizationName` or `givenName` and `surname`, (ii) `localityName` (to the extent such field is required under Section 7.1.4.2.2), (iii) `stateOrProvinceName` (to the extent required under Section 7.1.4.2.2), and (iv) `countryName` in the Subject field.

>   Such Certificates MUST also include either `organizationName` or both `givenName` and `surname`, `localityName` (to the extent such field is required under Section 7.1.4.2.2), `stateOrProvinceName` (to the extent required under Section 7.1.4.2.2), and `countryName` in the Subject field.  

Reviewer: ```suggestion
  Such Certificates MUST also include the following in the Subject field:
  (i) `organizationName`, or
  (ii) both `givenName` and `surname`, or
  (iii) both (i) and (ii)
  
  Additionally, Certificates MUST also include `localityName` (to the extent such field is required under Section 7.1.4.2.2), `stateOrProvinceName` (to the extent required under Section 7.1.4.2.2), and `countryName` in the Subject field.  
```

Author: Hrm, this seems substantially different (and formats terribly) that I'm going to reject this suggestion. I think, in the broader context, the problem you highlight regarding concern extends well beyond this particular sentence, and so I'm broadly OK with preserving the status quo here and seeing this tackled separately (perhaps as part of certificate profiles), given the inter-relationships.
